code/datums/mind.dm

```diff
@@ -1,739 +1,801 @@
-	if(!spell)
-		return
-	for(var/X in spell_list)
-		var/obj/effect/proc_holder/spell/S = X
-		if(istype(S, spell))
-			spell_list -= S
-			qdel(S)
-
-/datum/mind/proc/RemoveAllSpells()
-	for(var/obj/effect/proc_holder/S in spell_list)
-		RemoveSpell(S)
-
-/datum/mind/proc/transfer_martial_arts(mob/living/new_character)
-	if(!ishuman(new_character))
-		return
-	if(martial_art)
-		if(martial_art.base) //Is the martial art temporary?
-			martial_art.remove(new_character)
-		else
-			martial_art.teach(new_character)
-
-/datum/mind/proc/transfer_actions(mob/living/new_character)
-	if(current && current.actions)
-		for(var/datum/action/A in current.actions)
-			A.Grant(new_character)
-	transfer_mindbound_actions(new_character)
-
-/datum/mind/proc/transfer_mindbound_actions(mob/living/new_character)
-	for(var/X in spell_list)
-		var/obj/effect/proc_holder/spell/S = X
-		S.action.Grant(new_character)
-
-/datum/mind/proc/disrupt_spells(delay, list/exceptions = New())
-	for(var/X in spell_list)
-		var/obj/effect/proc_holder/spell/S = X
-		for(var/type in exceptions)
-			if(istype(S, type))
-				continue
-		S.charge_counter = delay
-		S.updateButtonIcon()
-		INVOKE_ASYNC(S, /obj/effect/proc_holder/spell.proc/start_recharge)
-
-/datum/mind/proc/get_ghost(even_if_they_cant_reenter, ghosts_with_clients)
-	for(var/mob/dead/observer/G in (ghosts_with_clients ? GLOB.player_list : GLOB.dead_mob_list))
-		if(G.mind == src)
-			if(G.can_reenter_corpse || even_if_they_cant_reenter)
-				return G
-			break
-
-/datum/mind/proc/grab_ghost(force)
-	var/mob/dead/observer/G = get_ghost(even_if_they_cant_reenter = force)
-	. = G
-	if(G)
-		G.reenter_corpse()
-
-
-/datum/mind/proc/has_objective(objective_type)
-	for(var/datum/antagonist/A in antag_datums)
-		for(var/O in A.objectives)
-			if(istype(O,objective_type))
-				return TRUE
-
-/mob/proc/sync_mind()
-	mind_initialize()	//updates the mind (or creates and initializes one if one doesn't exist)
-	mind.active = 1		//indicates that the mind is currently synced with a client
-
-/datum/mind/proc/has_martialart(string)
-	if(martial_art && martial_art.id == string)
-		return martial_art
-	return FALSE
-
-/mob/dead/new_player/sync_mind()
-	return
-
-/mob/dead/observer/sync_mind()
-	return
-
-//Initialisation procs
-/mob/proc/mind_initialize()
-	if(mind)
-		mind.key = key
-
-	else
-		mind = new /datum/mind(key)
-		SSticker.minds += mind
-	if(!mind.name)
-		mind.name = real_name
-	mind.current = src
-
-/mob/living/carbon/mind_initialize()
-	..()
-	last_mind = mind
-
-//HUMAN
-/mob/living/carbon/human/mind_initialize()
-	..()
-	if(!mind.assigned_role)
-		mind.assigned_role = "Unassigned" //default
-
-//AI
-/mob/living/silicon/ai/mind_initialize()
-	..()
-	mind.assigned_role = "AI"
-
-//BORG
-/mob/living/silicon/robot/mind_initialize()
-	..()
-	mind.assigned_role = "Cyborg"
-
-//PAI
-/mob/living/silicon/pai/mind_initialize()
-	..()
-	mind.assigned_role = ROLE_PAI
-	mind.special_role = ""
+/*	Note from Carnie:
```

Comment:
is it possible to fix whatever caused this I really don't want to manually check differences :(

Reply:
fucking web edits :(

Reply:
I got the master version of the file, re-applied the changes, still happens. No clue whats going on here. but I have a commit with all the changes to this file.

Reply:
https://github.com/tgstation/tgstation/pull/46913/commits/4500176cfa80a94fcb6da80cef56c566820f6134

Reply:
how can you see into my eyes

Reply:
This isn't resolved, the problem isn't just because it makes diffs hard to read for review. It also fucks with the file history

Reply:
Well can you help me fix it then because I tried for 15 min @ninjanomnom